Apps/Playground/macOS/ViewController.mm

```diff
@@ -3,15 +3,36 @@
 #import <Babylon/AppRuntime.h>
 #import <Babylon/Graphics.h>
 #import <Babylon/Plugins/NativeEngine.h>
-#import <Babylon/Plugins/NativeWindow.h>
 #import <Babylon/Polyfills/Window.h>
 #import <Babylon/Polyfills/XMLHttpRequest.h>
 #import <Babylon/ScriptLoader.h>
 #import <Shared/InputManager.h>
+#import <MetalKit/MetalKit.h>
 
 std::unique_ptr<Babylon::Graphics> graphics{};
 std::unique_ptr<Babylon::AppRuntime> runtime{};
 std::unique_ptr<InputManager<Babylon::AppRuntime>::InputBuffer> inputBuffer{};
+bool allgood = false;
```

Comment:
I've added this change and pushed into your branch instead of pushing to my repo. Sorry for that . I've added this boolean because EngineView rendering callback is being called just after setting the delegate. If I don't check the boolean, I get a crash in create index buffer. Loaded script creates index buffer (some default mesh?). This doesn't happen everytime. Sometimes, a crash, sometimes a dead lock with `finished` and sometimes, is works fine but a dead lock happens after few seconds. Feel free to discard my commit :)

Reply:
This has to do with the asynchrony mechanism that was originally supposed to be introduced in a separate change from this one, but has since been rolled into this PR as well.